File: Leet Code/Java/Problem507.java
/**
 * Problem507
 */
public class Problem507 {
    public static void main(String[] args) {
        boolean isPerfect = checkPerfectNumber(28);
        System.out.println(isPerfect);
    }

    static boolean checkPerfectNumber(int num) {
        int sum = 0;

        if (num == 1) {
            return false;
        }

        for(int i = 2; i < Math.sqrt(num); i++) {
            if (num % i == 0) {
                sum += i + (num / i);
            }
        }
        System.out.println("sum = " + sum);

        return sum == num;
    }
}

File: Leet Code/Python/problem_507.py
import math
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        sum = 1

        if(num == 1):
            return False

        count = 2
        while(count < math.sqrt(num)):
            if(num % count == 0):
                sum += count + (num / count)
            count += 1
        return sum == num

sl = Solution()
isperfect = sl.checkPerfectNumber(28)
print(isperfect)